app/models/asset.rb

```diff
@@ -230,6 +230,23 @@ def post_process_file(team = nil)
       # Update asset's estimated size immediately
       update_estimated_size(team)
     end
+
+    if is_image?
+      Asset.delay(queue: :assets).calculate_quality(id) if file.content_type == 'image/jpeg'
```

Comment:
I would try to implement it as custom processor https://github.com/thoughtbot/paperclip#custom-attachment-processors